server/src/main/java/org/opensearch/index/codec/composite/composite99/Composite99DocValuesWriter.java

```diff
@@ -205,7 +311,7 @@ private void mergeStarTreeFields(MergeState mergeState) throws IOException {
                     CompositeIndexValues compositeIndexValues = reader.getCompositeIndexValues(fieldInfo);
                     if (compositeIndexValues instanceof StarTreeValues) {
                         StarTreeValues starTreeValues = (StarTreeValues) compositeIndexValues;
-                        List<StarTreeValues> fieldsList = starTreeSubsPerField.getOrDefault(fieldInfo.getField(), Collections.emptyList());
+                        List<StarTreeValues> fieldsList = starTreeSubsPerField.getOrDefault(fieldInfo.getField(), new ArrayList<>());
                         if (starTreeField == null) {
                             starTreeField = starTreeValues.getStarTreeField();
                         }
```

Comment:
Can we add one test where star tree configuration is different across segments ? and merge .

Reply:
please elaborate on star tree configuration? is it the whole startreefield or just the StarTreeFieldConfiguration?

Reply:
As in the dims and metrics configuration being different for example. Even max leaf docs for instance. Anything that is different in star tree field should throw exception during merge.

Reply:
As discussed offline, taking in the next PR (beyond the scope of file formats as well).